wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/OnWorkspaceStartProjectInitializer.java

```diff
@@ -59,8 +60,10 @@ public void initialize()
     firePostInitializationHandlers();
   }
 
-  private void initializeRegisteredProjects() throws ServerException {
+  private void initializeRegisteredProjects()
+      throws ServerException, NotFoundException, ConflictException {
     for (ProjectConfig projectConfig : projectSynchronizer.getAll()) {
+      eventService.publish(new BeforeProjectInitializedEvent(projectConfig));
```

Comment:
Are events sent asynchronously? If yes, is our simple project migration code safe for it?

Reply:
These events sent synchronously and yes our code is safe.